app/main/controllers/privacy/RTMediaPrivacy.php

```diff
@@ -487,7 +487,15 @@ public function rt_privacy_settings_action() {
 
 			bp_core_add_message( $feedback, $feedback_type );
 			do_action( 'bp_core_general_settings_after_save' );
-			bp_core_redirect( bp_displayed_user_domain() . bp_get_settings_slug() . '/privacy/' );
+
+			// Function exists checks.
+			if ( function_exists( 'bp_get_requested_url' ) && function_exists( 'bp_displayed_user_domain' ) && function_exists( 'bp_get_settings_slug' ) ) {
```

Comment:
Just for information, if BuddyPress is installed then these are not their default function? can we check only one thing that if BuddyPress is there then this functions will be there, of only BuddyPress version checking.

Reply:
I think it's best to use `function_exists` check before using any 3rd party functions. If they change or remove a function, atleast our code will never give fatal error for undefined function.